### PR TITLE
Added an error for title length.

### PR DESCRIPTION
**Pull request by mmattbtw**
Pull request checklist:

- [x] `CHANGELOG.md` was updated, if applicable
- [x] Documentation in docs/ or install-docs/ was updated, if applicable
- [x] I have tested all changes

I did this because if a title is too long, pajbot responds saying "Banned words in title" or something along the lines.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -6,6 +6,7 @@ Remember to bring your dependencies up to date with `./scripts/venvinstall.sh` w
 
 - Minor: The `redirect_uri` config option under the `[twitchapi]` section is now optional with a sane default value: `https://<domain>/login/authorized`. (#1618)
 - Minor: Allow configuring streamer, bot, admin, and control hub using Twitch User IDs instead of Twitch User Logins. **Specifying user logins in the config has therefore now been deprecated and may stop working in a future release.** (#1590)
+- Minor: Added an error for a title being too long (#1638)
```
Comment (ALazyMeme) [inline CHANGELOG.md:9]: This should be a bugfix entry

**Review by RAnders00 — APPROVED** [review]

**Comment by RAnders00** [comment]
Is this 140 length documented somewhere? Or did you determine it empirically?

**Comment by mmattbtw** [comment]
@RAnders00 If you go into the Twitch dashboard and set your title, there is a character counter

**Review by ALazyMeme — APPROVED** [review]

**Comment by RAnders00** [comment]
What happens when you try and set a too-long title currently? I‘m mostly asking because if the API returns a specific error code (in the returned HTTP response body) indicating that the title was too long then we wouldn‘t need to hardcode the length of 140.

**Comment by mmattbtw** [comment]
@RAnders00 this is what currently happens.
image
Handled here: https://github.com/pajbot/pajbot/blob/6e59fd6e5eda317f7e9d862d226de8c06141774f/pajbot/modules/basic/stream_update.py#L138-L140

I looked into the Twitch documentation, and for the response codes it doesn't really get detailed, so I wouldn't know

**Comment by ALazyMeme** [comment]
> What happens when you try and set a too-long title currently? I‘m mostly asking because if the API returns a specific error code (in the returned HTTP response body) indicating that the title was too long then we wouldn‘t need to hardcode the length of 140.

Would the hardcoding of the limit not be better? Therefore we are not wasting API calls trying to set a title?

File: CHANGELOG.md [hunk 2]
```diff
@@ -6,6 +6,7 @@ Remember to bring your dependencies up to date with `./scripts/venvinstall.sh` w
 
 - Minor: The `redirect_uri` config option under the `[twitchapi]` section is now optional with a sane default value: `https://<domain>/login/authorized`. (#1618)
 - Minor: Allow configuring streamer, bot, admin, and control hub using Twitch User IDs instead of Twitch User Logins. **Specifying user logins in the config has therefore now been deprecated and may stop working in a future release.** (#1590)
+- Bugfix: Added an error for a title being too long (#1638)
```
Comment (pajlada) [inline CHANGELOG.md:9]:
```suggestion
- Bugfix: Added an error for a title being too long. (#1638)
```

**Review by pajlada — APPROVED** [review]
Since we can't differentiate the 400 status code between the various errors, doing a client-side check like this makes sense to me :+1:

**Pull request merged**